File: plugins/widget-bridge/android/src/main/java/app/ok2merge/bridge/WidgetBridge.java
package app.ok2merge.bridge;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "WidgetBridge")
public class WidgetBridge extends Plugin {
  @PluginMethod
  public void writeSnapshot(PluginCall call) {
    String json = call.getString("json", "{}");
    getContext()
      .getSharedPreferences("ok2merge_widget", 0)
      .edit()
      .putString("snapshot", json)
      .apply();
    call.resolve();
  }
}

File: plugins/widget-bridge/android/src/main/java/app/ok2merge/bridge/WidgetBridge.kt
package app.ok2merge.bridge

import com.getcapacitor.CapacitorPlugin
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod

@CapacitorPlugin(name = "WidgetBridge")
class WidgetBridge : Plugin() {
  @PluginMethod
  fun writeSnapshot(call: PluginCall) {
    context
      .getSharedPreferences("ok2merge_widget", 0)
      .edit()
      .putString("snapshot", call.getString("json") ?: "{}")
      .apply()
    call.resolve()
  }
}
